### PR TITLE
Upgrade service to use latest dependencies

### PR DESCRIPTION
**Pull request by nshumoogum**
### What

- Go version updated to 1.15
- App dependencies upgraded to use latest stable versions of libraries
  - dp-net replaces dp-rchttp
  - dp-api-clients-go replacing go-ns functionality
  - dp-kafka/v2 replacing dp-kafka
- Remove embedded kafka audit in API
- Fix make audit command

### How to review

- Check unit tests pass
- Check health endpoint continues to check external dependencies
- Sanity check code changes

### Who can review

Anyone

### CODE REVIEW

**Review by redhug1 — COMMENTED** [review]
Line 28 has '-race' in it twice

File: main.go [hunk 1]
```diff
@@ -43,45 +40,30 @@ func main() {
...
-		log.Event(ctx, "failed to start up, unable to connect to elastic search instance", log.FATAL, log.Error(err), log.Data{"http_status": status})
-		os.Exit(1)
-	}
 
-	producerChannels := kafka.CreateProducerChannels()
-	hierarchyBuiltProducer, err := kafka.NewProducer(ctx, cfg.Brokers, cfg.HierarchyBuiltTopic, cfg.KafkaMaxBytes, producerChannels)
+	// producerChannels := kafka.CreateProducerChannels()
+	// hierarchyBuiltProducer, err := kafka.NewProducer(ctx, cfg.Brokers, cfg.HierarchyBuiltTopic, cfg.KafkaMaxBytes, producerChannels)
+	// exitIfError(ctx, err, "error creating kafka hierarchyBuiltProducer")
+	// hierarchyBuiltProducer.Channels().LogErrors(ctx, "error received from hierarchy built kafka producer, topic: "+cfg.HierarchyBuiltTopic)
```
Comment (CarlHembrough) [inline main.go:49]: Remove the commented code?

File: main.go [hunk 2]
```diff
@@ -136,20 +116,6 @@ func configureHealthChecks(ctx context.Context,
...
 	}
 
-	if cfg.HasPrivateEndpoints {
-		if err = hc.AddCheck("Kafka Audit Producer", auditProducer.Checker); err != nil {
-			log.Event(ctx, "error adding check for kafka audit producer", log.ERROR, log.Error(err))
-			hasErrors = true
-		}
-
-		// zebedee is used only for identity checking
-		zebedeeClient := zebedee.New(cfg.AuthAPIURL)
```
Comment (CarlHembrough) [inline main.go:146]: The Zebedee health check has been removed - is that right? The config is still there and I can't see why it would be removed.

**Review by CarlHembrough — CHANGES_REQUESTED** [review]
Changes look good generally - please check comments.

File: main.go (hunk 1 above)
Comment (nshumoogum) [reply main.go:49]: Lines 46-49 are not needed as these are replaced with line 51+. I should reinstate line 49 after I've created a new kafka producer for hierarchy topic

**Review by CarlHembrough — APPROVED** [review]
LGTM 👍

**Pull request merged**